### PR TITLE
Support template watch

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

**Comment by jimmidyson** [comment]
@mfojtik Any chance of taking a look at this quick change?

File: pkg/template/registry/etcd/etcd.go [hunk 1]
```diff
@@ -59,3 +60,8 @@ func (r *REST) NewList() runtime.Object {
 func (r *REST) List(ctx kapi.Context, label labels.Selector, field fields.Selector) (runtime.Object, error) {
 	return r.Etcd.ListPredicate(ctx, registry.MatchTemplate(label, field))
 }
+
+// Watch begins watching for new, changed, or deleted image streams.
```
Comment (mfojtik) [inline pkg/template/registry/etcd/etcd.go]: comment seems to be copied from image streams :-)

**Comment by mfojtik** [comment]
LGTM

@smarterclayton can you have second look?

File: pkg/template/registry/etcd/etcd.go (hunk 1 above)
Comment (jimmidyson) [inline pkg/template/registry/etcd/etcd.go]: I don't what you mean ;) Fixing

Comment (mfojtik) [inline pkg/template/registry/etcd/etcd.go]: @jimmidyson no worries, we're all doing it :-)

**Comment by smarterclayton** [comment]
@ncdc

**Comment by ncdc** [comment]
LGTM. Do we want or need test cases in this PR or a follow-up?

**Comment by jimmidyson** [comment]
@ncdc I was copy/pasting from elsewhere so copied the no test approach from watching image streams (test commented out: https://github.com/openshift/origin/blob/master/pkg/image/registry/imagestream/etcd/etcd_test.go#L837) ;) Not sure why it's commented out?

Unless I've missed an e2e test perhaps?

**Comment by ncdc** [comment]
It's mostly commented out because of a big refactor and the fact that watching should be fully functional when using generic etcd. If you only support watching and matching on certain fields or labels, that might make more sense. For now it's probably ok.

[merge]

**Pull request merged**